stdlib/REPL/src/REPLCompletions.jl

```diff
@@ -583,42 +597,105 @@ function complete_any_methods(ex_org::Expr, callee_module::Module, context_modul
+                # argument types
+                push!(args_ex, isexpr(ex, :...) ? Vararg{Any} : Any)
             else
                 push!(args_ex, get_type(get_type(ex, context_module)..., default_any))
             end
         end
     end
-    return args_ex, kwargs_ex
+    return args_ex, Set{Symbol}(kwargs_ex), kwargs_flag
+end
+
+function complete_methods_args(funargs::Vector{Any}, ex_org::Expr, context_module::Module, default_any::Bool, allow_broadcasting::Bool)
+    if allow_broadcasting && ex_org.head === :. && ex_org.args[2] isa Expr
+        return detect_args_kwargs((ex_org.args[2]::Expr).args, context_module, default_any, Val(true))
+    end
+    return detect_args_kwargs(funargs, context_module, default_any, Val(false))
 end
 
-function complete_methods!(out::Vector{Completion}, @nospecialize(funct), args_ex::Vector{Any}, kwargs_ex::Bool, max_method_completions::Int)
+function complete_methods!(out::Vector{Completion}, @nospecialize(funct), args_ex::Vector{Any}, kwargs_ex::Set{Symbol}, max_method_completions::Int, exact_narg::Bool)
     # Input types and number of arguments
-    t_in = Tuple{funct, args_ex...}
+    num_splat = 0 # number of splat arguments in args_ex
+    args_ex_onevararg = copy(args_ex) # args_ex_onevararg contains at most one Vararg, put in final position
+    for i in 1:length(args_ex)
+        arg = args_ex[i]
+        if Base.isvarargtype(arg)
+            num_splat += 1
+            num_splat > 1 && continue
+            args_ex_onevararg[i] = Vararg{Any}
+            resize!(args_ex_onevararg, i)
+        end
+    end
+    t_in = Tuple{funct, args_ex_onevararg...}
     m = Base._methods_by_ftype(t_in, nothing, max_method_completions, Base.get_world_counter(),
         #=ambig=# true, Ref(typemin(UInt)), Ref(typemax(UInt)), Ptr{Int32}(C_NULL))
     if m === false
         push!(out, TextCompletion(sprint(Base.show_signature_function, funct) * "( too many methods to show )"))
     end
     m isa Vector || return
     for match in m
-        # TODO: if kwargs_ex, filter out methods without kwargs?
+        if exact_narg
+            # If there is a semicolon, the number of non-keyword arguments in the
+            # call cannot grow. It must thus match that of the method.
+            meth_nargs = match.method.nargs - 1 # remove the function itself
+            exn_args = length(args_ex)
+            if meth_nargs != exn_args
+                # the match may still hold if some arguments are slurped or splat
+                num_slurp = count(Base.isvarargtype, Base.unwrap_unionall(match.method.sig).types)
+                num_slurp == 0 && num_splat == 0 && continue
+                if num_slurp == 0
+                    meth_nargs ≥ exn_args - num_splat || continue
+                elseif num_splat == 0
+                    exn_args ≥ meth_nargs - num_slurp || continue
+                end
+            end
+        end
+
```

Comment:
This is not how we define these values (it is stored in `match.method.isva`), and it won't compute it reliably. But we already computed this when creating `args_ex`, when deciding whether to append `Vararg{Any}`, so none of this code seems necessary. Otherwise the list you are filtering would already be wrong, and it would already be hopeless to fix.

```suggestion
```